feat: add site settings and google drive folder setting

jeremydw opened the pull request:
This PR adds the ability to manage site settings (settings associated with the site, not the user) in the CMS UI. It creates a framework for adding more settings over time.

For now, we just add one setting: "Google Drive folder". This allows site admins to specify a Google Drive folder that will be used by the localization process. If specified, when the localization process creates Google Sheets, it will place them in the Google Drive folder set in the site setting, rather than the invoker's "My Drive".

The settings are stored in Firestore, same as roles, meaning only admins can change them.

```
Projects/:project/settings/:key
```

<img width="1832" height="1272" alt="image" src="https://github.com/user-attachments/assets/2986e6af-5042-4c11-9a39-589f1cfe2831" />

<img width="2204" height="1840" alt="image" src="https://github.com/user-attachments/assets/086f7b1c-bf56-4c20-ad37-bdd14724156b" />

<img width="678" height="354" alt="image" src="https://github.com/user-attachments/assets/174367d3-da85-4c74-ab3d-ce6f92359866" />

stevenle commented:
when a new spreadsheet is created, the ACLs from the project are added to the new spreadsheet. when spreadsheets are created within a google drive folder, do you think we can skip that step (saves rpc time, plus sometimes it's buggy if a user account is deleted)

jeremydw commented:
ah yeah; that's a good thing to fix. how about this:

1. we could add a checkbox to the "create sheet" flow that says "Share Google Sheet with participants" and, if checked, will run through the current sharing flow
2. if the site has a Google Drive folder configured, that checkbox can be unchecked by default
3. if the site does _not_ have a Google Drive folder configured, that checkbox would remain checked

how about that?

stevenle commented:
checkbox sounds good to me!

jeremydw commented:
Made the following changes:

1. Added checkbox to toggle the sharing upon creation
2. Fixed a subtle bug with the "Open Google Sheet" button
3. Disabled the fields after the sheet is created

<img width="687" height="413" alt="image" src="https://github.com/user-attachments/assets/8c0a29cd-c4ad-44f9-b889-4aae83c91262" />

<img width="679" height="405" alt="image" src="https://github.com/user-attachments/assets/d68b1980-2e5b-44f6-b31d-799fa0be7e4d" />
